### ISSUE TITLE
Expressing trait type restrictions via protocol?

### ISSUE THREAD

**Issue by antonselyanin** [krzysztofzablocki/Traits#7]
As I understand, if I want `Font` trait to support `UITextView` I have to modify `Font` source code: add `UITextView.self` to `restrictedTypes` array and modify `apply` method.

IMO, if target type restrictions were expressed via a protocol, it would be easier to add new target classes to existing traits.

**Comment by krzysztofzablocki** [comment]
example code? Do you mean having something like FontModifiable that would express protocol for modifying them and then conforming as extensions?

**Comment by antonselyanin** [comment]
Yes, it would be easier to reuse traits if each trait had such `{Trait}Modifiable` protocol.

If `Font` trait had `FontModifiable` protocol for targets, I could conform `UITextView` to this protocol and use the trait.

**Comment by krzysztofzablocki** [comment]
thats an idea for a PR, do you want to try contributing it?

**Comment by antonselyanin** [comment]
Sure!
